### PR TITLE
Teacher tool top level doc page

### LINKED ISSUE #5603: Teacher Tool Testing Documentation
We should do some documentation around the teacher tool, public facing and just to track features for testing.

## Overview
The teacher tool is a mechanism for constructing a checklist of requirements and running that list automatically against projects in quick succession. This allows teachers to build a checklist, then easily evaluate any number of projects based on that checklist. Projects must be evaluated one at a time, but with auto-run enabled, you can update the loaded project by providing a new share link, at which point the rules will automatically be re-run on the new project.

You can access the beta version of the teacher tool here: https://microbit.makecode.com/beta--eval

_Side note: these notes use rubric/checklist interchangeably. Going forward, "checklist" is the correct term. Tool UI strings still need updating to account for this._

### Creating, Editing, and Running a Checklist
1. User can create a new rubric using the "New Rubric" card (or from the "..." action menu)
   - If there is already an "in progress" checklist, a warning will appear asking if you are okay with overwriting it.
   - <details>
      <summary>Screenshots</summary>

      image
      image

      </details>

2. User can give the checklist a name
   - <details>
      <summary>Screenshots</summary>
      
      image

      </details>

3. User can add criteria from the catalog using the "Add Criteria" button.
   - Some criteria (like `[block] used [count] times`) can be added multiple times, others (like `Read a GPIO pin` can only be added once)
   - <details>
      <summary>Screenshots</summary>
      
      image
      image

      </details>

4. User can remove criteria using the trash button
   - <details>
      <summary>Screenshots</summary>
      
      image

      </details>

5. User can fill in parameters
   - Numeric parameters have a small input and only allow number inputs
   - String parameters can have medium and long sized inputs
   - Block parameters should open a block-picker modal
   - Empty parameters appear in an error state until they have values
   - <details>
      <summary>Screenshots</summary>
      
      image
      image
      image

      </details>

6. User can load a project into the project view by pasting in a share link or share id
   - The project should load in read-only mode
   - The project title should appear at the top of the project view
   - <details>
      <summary>Screenshots</summary>
      
      image
      image

      </details>
      
7. With a project loaded, user can run the checklist and see results by clicking the "Run" button
   - Button is disabled without loaded project
   - <details>
      <summary>Screenshots</summary>
      
      image
      image

      </details>

### Editing Results
1. User can add feedback/notes using the "Add Notes" button
   - The feedback box should resize to fit its content
   - Feedback is persisted if the user re-runs the rules using the "Run" button
   - <details>
      <summary>Screenshots</summary>
      
      image
      image
      image

      </details>
4. User can edit an outcome using the provided dropdown
   - <details>
      <summary>Screenshots</summary>
      
      image
      image
      
      </details>

### Result Clearing and Auto-Run
1. If auto-run is **disabled**, a result's outcome (i.e. "Looks good", "Needs work", etc...) should be set to "Not started" automatically when any of any of the following conditions are met:
   - It is newly added (should default to the "Not Started" state)
   - A parameter in a rule is changed (only the affected rule should enter the "Not Started" state)
   - The loaded project changes (all rules should be set to "Not started")
   
2. Auto-run can be toggled on/off using the button in the menu.
   - <details>
      <summary>Screenshots</summary>
      
      image
      
      </details>
3. If auto-run is **enabled**, any rules that enter the "Not started" state from the above conditions should immediately and automatically get re-run and have their results updated.

### Loading/Importing/Exporting Checklists
1. User can open pre-built rubrics from the home page
   - Will ask for overwrite confirmation if the user already has an in-progress checklist.
   - <details>
      <summary>Screenshots</summary>
      
      image
      
      </details>
2. User can export a checklist from the vertical "..." menu near the "auto-run" button. This will download a json file.
   - <details>
      <summary>Screenshots</summary>
      
      image
      image

      </details>
3. User can import a rubric from a file using the same "..." menu, or from the card on the welcome page.
   - Rubric file can be selected via "Browse" or dropped directly into the popup
   - Will ask for overwrite confirmation if the user already has an in-progress checklist.
   - <details>
      <summary>Screenshots</summary>
      
      image
      image
      image
      image

      </details>
      
### Other
- If user refreshes the page (or closes/re-opens the browser), their checklist should be preserved.
- User can click the print button to create a version of the results with the outcome and feedback visible, but other UI elements hidden (currently broken).
   - <details>
      <summary>Screenshots</summary>
      
      image

      </details>
- The checklist-view/project-view splitter can be resized. It can also be reset to 50/50 split with double-click.
   - <details>
      <summary>Screenshots</summary>
      
      image
      image

      </details>

### LINKED ISSUE #5622: Teacher Tools: Shows 'Sorry, the page is missing' when click 'Learn More'
**Describe the bug**
Teacher Tools: Shows 'Sorry, the page is missing' when click **'Learn More'**

**Steps to reproduce the behavior**
1.Navigate to [MakeCode Teacher Tool (microbit.org)](https://makecode.microbit.org/beta--eval).
2.Click **'Learn More'** and observe.
 
**Actual behavior:**
Shows 'Sorry, the page is missing' in new tab.
 
**Expected behavior:**
Open the help docs with no error.
 
**Screen Recording:**
Learnmore

**Additional context:**
1.OS: Windows11
2.makecode.microbit,org version: 7.0.6
3.Microsoft MakeCode version: 10.0.18

### PR DESCRIPTION
**Pull request by ganicke**
Here an overview of the Teacher Tool features. This kind of a rip/redo of the info from #5603. This could double as  the feature testing guide.

Is a single page guide sufficient or spread this out to multiple pages? I'm assuming the term "rubric" will in all places be replaced with "checklist"? Some of the images will need a re-capture to reflect updates. Any comments or descriptive additions are welcome.

Closes #5603 
Closes #5622

### CODE REVIEW

File: docs/teachertool.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
+# Teacher Tool
```
Comment (thsparks) [inline docs/teachertool.md:1]: The official name will be the "MakeCode Code Evaluation Tool", or for the purposes of this doc, probably just "Code Evaluation Tool"

File: docs/teachertool.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+The [Teacher Tool]( https://microbit.makecode.com/beta--eval) is a mechanism for constructing a checklist of requirements for an assignment and running that list automatically against projects in quick succession. This allows teachers to build a checklist, then easily evaluate any number of projects based on that checklist. Projects are evaluated one at a time, but with auto-run enabled, you can update the loaded project by providing a new share link, at which point the rules will automatically be re-run on the new project.
+
+[Teacher tool video/gif goes here]
+
+## Teacher Tool Features
+
+### Creating, Editing, and Running a Checklist
+
+#### 1. Creating a new rubric
```
Comment (thsparks) [inline docs/teachertool.md:13]: You mentioned this in your PR description already, but yeah, we'll want to replace "Rubric" with "Checklist"

File: docs/teachertool.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+A user creates a new rubric using the **New Rubric** card (or from the **...** action menu). If there is already an "in progress" checklist, a warning will appear asking if it is okay to overwrite it.
```
Comment (thsparks) [inline docs/teachertool.md:15]:
```suggestion
Create a new checklist using the **New Checklist** card. If there is already an "in progress" checklist, a warning will appear asking if it is okay to overwrite it.
```

Just rephrasing in a more "instructional/declarative" capacity, as opposed to referencing the "user".

File: docs/teachertool.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+Some criteria (like `[block] used [count] times`) can be added multiple times, others (like `Read a GPIO pin` can only be added once).
+
+![Criteria items](/static/teachertool/criteria-items.png)
+
+#### 4. Fill in Parameters
+
+Parameters for the criteria item are filled in. Here are the parameter types:
+
+- **Numeric** parameters have a small input and only allow number inputs.
```
Comment (thsparks) [inline docs/teachertool.md:41]: I think listing the parameter types may be TMI, but happy to defer to other folks' perspectives on this.

File: docs/teachertool.md [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+- **Block** parameters should open a block-picker modal.
+- **Empty** parameters appear in an error state until they have values.
+      
+![Criteria parameters 1](/static/teachertool/parameters-1.png)
+
+Parameter options are displayed and then selected.
+
+![Criteria parameters 2](/static/teachertool/parameters-2.png)
+      
+![Criteria parameters 3](/static/teachertool/parameters-3.png)
```
Comment (thsparks) [inline docs/teachertool.md:52]: Unfortunately the visuals changed recently so we'll have to redo some screenshots eventually, but not a huge issue. I think this is fine as-is for now.

**Review by thsparks — APPROVED** [review]
Thank you so much for writing all of this up! We'll have to tweak a few things over time, but this is a great foundation!

File: docs/teachertool.md (hunk 5 above)
Comment (ganicke) [reply docs/teachertool.md:60]: We'll revisit the screen shots when everything is final and maybe insert a gif/vid of an auto-run for one of the pre-built checklists.

**Comment by ganicke** [comment]
This will fill the need for a destination doc for `../docs/teachertool`. Will update the screencaps at release time.

**Pull request merged**